mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -376,35 +456,121 @@ template <typename T> struct MILARewritePattern : public OpRewritePattern<T> {
         laGeneric->moveBefore(twcopy);
 
         // 2.5. Replace twcopy inputs with vector from la result vgpr
-        auto vload0 =
-            b.create<vector::LoadOp>(loc, regVecType, laOutRegs, coords0);
-        auto vload1 =
-            b.create<vector::LoadOp>(loc, regVecType, laOutRegs, coords1);
+        auto vload0 = b.create<vector::LoadOp>(loc, regVecType, laOutRegs, c0);
+
         twcopys.back()->setOperand(0, vload0);
-        twcopys.front()->setOperand(0, vload1);
 
         // 2.6. Reset twcopy output to point to old laGeneric output
         auto mrReshape =
             transforms.front().getDefiningOp<miopen::TransformOp>();
         mrReshape->setOperand(0, out);
+
+        return success();
+        }
       }
+
+      return fail;
     }
+  };
+
+  //===----------------------------------------------------------------------===//
+  // ThreadwiseCopyV2 Conversion in Fusion.
+  //===----------------------------------------------------------------------===//
+
+  struct ThreadwiseCopyV2RewritePattern
```

Comment:
So,  xdlops fusion consists of two steps. First the fusion logic fuses and it includes illegal threadwise copy v2 (mem ->vector) and this pattern only convert that copy.
At least, this needs more comment to explain. And it might be reworked as a function if there is a strong reason not to do this way.

Reply:
Could this have been a function instead of a rewrite?